api/v1alpha1/externaldns_types.go

```diff
@@ -18,41 +18,412 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// +genclient
```

Comment:
I expected this to be cluster-scoped.  You could have multiple instances in a single cluster, but control of DNS seems like a cluster-admin scoped power.

Were you thinking of namespace scoping?  If so, what sort of topology do you have in mind?  Something more akin to hypershift?

Reply:
Changed to cluster-scoped, since that makes more sense than namespace scoped. Thanks!